src/cpx_newcbs.jl

```diff
@@ -0,0 +1,211 @@
+mutable struct CallbackContext
+    model::Model
+    context::Ptr{Cvoid}
+end
+
+function callback_wrapper(context::Ptr{Cvoid},
+                          context_id::Clong,
+                          user_handle::Ptr{Cvoid})
+    (model, user_callback) = unsafe_pointer_to_objref(user_handle)::Tuple{Model, Function}
+    # Run the `user_callback`.
+    user_callback(CallbackContext(model, context), context_id)
+    # Exit normally.
+    return Cint(0)
+end
+
+"""
+    cpx_callbacksetfunc(model::Model, context_mask::Clong, user_callback::Function)
+
+Set the general callback function of model to `user_callback`.
+
+`user_callback` is a function that takes two arguments. The first is a
+`CallbackContext` instance, and the second is a `Clong` indicating where the
+callback is being called from.
+
+## Examples
+
+    function my_callback(cb_context::CPLEX.CallbackContext, context_id::Clong)
+        CPLEX.cpx_callbackabort(cb_context)
+    end
+    CPLEX.cpx_callbacksetfunc(model, Clong(0), my_callback)
+"""
+function cpx_callbacksetfunc(model::Model, context_mask::Clong, user_callback::Function)
+    c_callback = @cfunction(
+        callback_wrapper, Cint, (Ptr{Cvoid}, Clong, Ptr{Cvoid})
+    )
+    user_handle = (model, user_callback)::Tuple{Model, Function}
+    ret = @cpx_ccall(callbacksetfunc,
+        Cint,
+        (Ptr{Cvoid}, Ptr{Cvoid}, Clong, Ptr{Cvoid}, Any),
+        model.env, model.lp, context_mask, c_callback, user_handle
+    )
+    if ret != 0
+        throw(CplexError(model.env, ret))
+    end
+    # We need to keep a reference to the callback function so that it isn't
+    # garbage collected.
+    model.callback = user_handle
+    return
+end
+
+# int CPXXcallbackabort( CPXCALLBACKCONTEXptr context )
+function cpx_callbackabort(callback_data::CallbackContext)
```

Comment:
Okay @chengg04, I've wrapped `callbackabort` here so you can see what I did. 

Now we just need to do the same for the other functions here:
https://www.ibm.com/support/knowledgecenter/en/SSSA5P_12.8.0/ilog.odms.cplex.help/refcallablelibrary/groups/homepagecallbacks.html